cmd/tools/vdoc/theme/doc.js

```diff
@@ -1,235 +1,204 @@
-                    var tocHeight = docToc.clientHeight;
-                    var scrollTop = docToc.scrollTop;
-                    if ((document.body.getBoundingClientRect()).top < scrollPos && scrollTop < link.offsetTop - 10) {
-                        docToc.scrollTop = link.clientHeight + link.offsetTop - tocHeight + 10;
-                    } else if (scrollTop > link.offsetTop - 10) {
-                        docToc.scrollTop = link.offsetTop - 10;
-                    }
-                }
-                break;
-            }
-        }
-        scrollPos = (document.body.getBoundingClientRect()).top;
-    });
+	const sections = document.querySelectorAll('section');
+	const sectionPositions = Array.from(sections).map((section) => section.offsetTop);
+	let scrollPos = 0;
+	window.addEventListener('scroll', () => {
+		const toc = document.querySelector('.doc-toc');
+		// Reset classes
+		toc.querySelectorAll('a[class="active"]').forEach((link) => link.classList.remove('active'));
+		// Set current menu link as active
+		let scrollPosition = document.documentElement.scrollTop || document.body.scrollTop;
+		for (const [i, position] of sectionPositions.entries()) {
+			if (position >= scrollPosition) {
+				const section = sections[i];
+				const link = toc.querySelector('a[href="#' + section.id + '"]');
+				if (link) {
+					link.classList.add('active');
+					const tocHeight = toc.clientHeight;
+					const scrollTop = toc.scrollTop;
+					if (
+						document.body.getBoundingClientRect().top < scrollPos &&
+						scrollTop < link.offsetTop - 10
+					) {
+						toc.scrollTop = link.clientHeight + link.offsetTop - tocHeight + 10;
+					} else if (scrollTop > link.offsetTop - 10) {
+						toc.scrollTop = link.offsetTop - 10;
+					}
+				}
+				break;
+			}
+		}
+		scrollPos = document.body.getBoundingClientRect().top;
+	});
 }
 
 function setupMobileToggle() {
-    var toggle = document.getElementById('toggle-menu');
-    toggle.addEventListener('click', function (_) {
-        var docNav = document.querySelector('.doc-nav');
-        var isHidden = docNav.classList.contains('hidden');
-        docNav.classList.toggle('hidden');
-        var search = document.querySelector('.doc-nav > .search');
-        console.log(search);
-        var searchHasResults = search.classList.contains('has-results');
-        if (isHidden && searchHasResults) {
-            search.classList.remove('mobile-hidden');
-        } else {
-            search.classList.add('mobile-hidden');
-        }
-        var content = document.querySelector('.doc-nav .content');
-        content.classList.toggle('hidden');
-        content.classList.toggle('show');
-    });
+	document.getElementById('toggle-menu').addEventListener('click', () => {
+		const docNav = document.querySelector('.doc-nav');
+		const isHidden = docNav.classList.contains('hidden');
+		docNav.classList.toggle('hidden');
+		const search = docNav.querySelector('.search');
+		// console.log(search);
```

Comment:
Remove this console.log

Reply:
:+1: I'll do it when adding following changes. It was there so I just commented it out here.

Thanks for the review ! :heart: